src/clients/c++/request_http.cc

```diff
@@ -1130,6 +1233,10 @@ InferHttpContextImpl::PreRunProcessing(std::shared_ptr<Request>& request)
   list = curl_slist_append(list, "Expect:");
   list = curl_slist_append(list, "Content-Type: application/octet-stream");
   list = curl_slist_append(list, infer_request_str_.c_str());
+  for (const auto& pr : headers_) {
+    std::string hdr = pr.first + ": " + pr.second;
```

Comment:
What will happen if users pass in their own infer request header? Should we add check to prevent them from doing so?

Reply:
I'll check for it during initialization of the context